### PR TITLE
Added schuerch dataset specific transforms

### PR DESCRIPTION
**Pull request by JLrumberger**
## What is the purpose of this PR?
This PR adds dataset specific transformations for the schuerch dataset.

## How did you implement your changes?
I added a function `exclude_classes` that get's rid of the 'dirt' semantic class and also get's rid of the associated segments in the instance segmentation map. I added a function `transform_semantic_mask` that maps the semantic masks from the original schuerch labels to a reduced set of labels via this mapping (keys original classes, values new classes):

```python
ct_rename_dict = {
    "background": "Background",
    "B cells": "B cells",
    "CD11b+ monocytes": "Macrophages/Monocytes",
    "CD11b+CD68+ macrophages": "Macrophages/Monocytes",
    "CD11c+ DCs": "Dendritic cells",
    "CD163+ macrophages": "Macrophages/Monocytes",
    "CD3+ T cells": "T cells",
    "CD4+ T cells": "T cells",
    "CD4+ T cells CD45RO+": "T cells",
    "CD4+ T cells GATA3+": "T cells",
    "CD68+ macrophages": "Macrophages/Monocytes",
    "CD68+ macrophages GzmB+": "Macrophages/Monocytes",
    "CD68+CD163+ macrophages": "Macrophages/Monocytes",
    "CD8+ T cells": "T cells",
    "NK cells": "NK cells",
    "Tregs": "T cells",
    "adipocytes": "Adipocytes",
    "granulocytes": "Granulocytes",
    "immune cells": "Other cells",
    "immune cells / vasculature": "Other cells",
    "lymphatics": "Vasculature/Lymphatics",
    "nerves": "Nerves",
    "plasma cells": "Plasma cells",
    "smooth muscle": "Smooth muscle",
    "stroma": "Stroma",
    "tumor cells": "Tumor cells",
    "tumor cells / immune cells": "Other cells",
    "undefined": "Other cells",
    "vasculature": "Vasculature/Lymphatics",
}
```
Which has the following resulting mapping dict:

```python
coarse_mapping = {
    0: 'Background',
    1: 'B cells',
    3: 'Macrophages/Monocytes',
    4: 'Adipocytes',
    5: 'Dendritic cells',
    6: 'T cells',
    7: 'Granulocytes',
    8 : 'NK cells',
    9 : 'Nerves',
    10: 'Plasma cells',
    11: 'Smooth muscle',
    12: 'Stroma',
    13: 'Tumor cells',
    14: 'Vasculature/Lymphatics',
    15: 'Other cells',
}
```

## How did you test your changes?
Exhaustivly tested both functions on mock data.

## Remaining issues
None

### CODE REVIEW

File: src/bio_image_datasets/schuerch_dataset.py [hunk 1]
```diff
@@ -164,3 +165,102 @@ def get_sample_names(self):
...
+    5: 'Dendritic cells',
+    6: 'T cells',
+    7: 'Granulocytes',
+    8 : 'NK cells',
+    9 : 'Nerves',
+    10: 'Plasma cells',
+    11: 'Smooth muscle',
+    12: 'Stroma',
+    13: 'Tumor cells',
+    14: 'Vasculature/Lymphatics',
```
Comment (NoraKrbr) [inline src/bio_image_datasets/schuerch_dataset.py:209]: why did you put vasculature & lymphatics together? is there a biological reason?

File: src/bio_image_datasets/schuerch_dataset.py [hunk 2]
```diff
@@ -164,3 +165,102 @@ def get_sample_names(self):
...
+        semantic_mask (np.ndarray): The semantic mask.
+        exclude_classes (list): List of classes to exclude.
+        instance_mask (np.ndarray): The instance mask.
+    Returns:
+        np.ndarray: The updated semantic mask.
+        optional np.ndarray: The updated instance mask.
+    """
+    # Make a copy of the input masks
+    semantic_mask = copy(semantic_mask)	
+    if instance_mask is not None:
```
Comment (NoraKrbr) [inline src/bio_image_datasets/schuerch_dataset.py:183]: is there not always an instance mask?

File: src/bio_image_datasets/schuerch_dataset.py (hunk 1 above)
Comment (JLrumberger) [reply src/bio_image_datasets/schuerch_dataset.py:209]: They look super similar (white tubes consisting of endothelial cells), thus I put them together in one class. You may be able to differentiate them depending on the content of the vessel (i.e. red blood cells should only be in vasculature not in lymphatic vessels).

File: src/bio_image_datasets/schuerch_dataset.py (hunk 2 above)
Comment (JLrumberger) [reply src/bio_image_datasets/schuerch_dataset.py:183]: Nope, for the segpath dataset we only have semantic masks, no instance masks.

**Pull request merged**